underfs/s3a/src/main/java/alluxio/underfs/s3a/AlluxioS3Exception.java

```diff
@@ -0,0 +1,75 @@
+package alluxio.underfs.s3a;
+
+import alluxio.exception.AlluxioRuntimeException;
+
+import com.amazonaws.AmazonClientException;
+import com.amazonaws.AmazonServiceException;
+import com.google.protobuf.Any;
+import io.grpc.Status;
+
+import java.net.HttpURLConnection;
+
+/**
+ * Alluxio exception for s3.
+ */
+public class AlluxioS3Exception extends AlluxioRuntimeException {
+
+  /**
+   * Converts an AmazonClientException to a corresponding AlluxioS3Exception.
+   * @param cause aws s3 exception
+   * @return alluxio s3 exception
+   */
+  public static AlluxioS3Exception from(AmazonClientException cause) {
```

Comment:
I'd like to return the [S3 error code](https://docs.aws.amazon.com/AmazonS3/latest/API/ErrorResponses.html#ErrorCodeList) in the Alluxio error message.

Is it possible to receive [AmazonS3Exception](https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/services/s3/model/AmazonS3Exception.html) instead of `AmazonClientException`?

Reply:
I just think we only have two types of exceptions that would happen: one is [AmazonClientException] indicates that a severe problem occurred inside the Java client code. And another is AmazonS3Exception. So I think it might be better to catch AmazonClientException and handle the s3 exception within it. 
I put the error code into the error message right now but I'm open to other access method like getter

Reply:
Ah I didn't know `AmazonClientException` wrapped `AmazonS3Exception`. So long as the error code is captured in the error message I'm fine with whatever approach. thanks!